### ISSUE TITLE
Yard graph with alternate db fails

### ISSUE THREAD

**Issue by japgolly** [lsegal/yard#583]
Similar to #580, `yard graph --db <alternate-dir>` doesn't work either.

Easily solved though. I'll submit 2 separate pull requests, one big-picture; one isolated. Choose whichever makes you happier. :)

**Note though:** I strongly dislike having Yard contribute two directories of purely generated data to my project root (`.yardoc` and `doc`). Other projects like to do the same and the end result is about 500 (possibly exaggerated) directories, some valuable, some transient - it all becomes very messy and strains minds and the housekeeping cost.
I therefore love that Yard supports me moving this stuff around - it just seems like the functionality hasn't been very well tested yet. THUSLY I'll likely be raising more issues around this so...
**TL;DR** I recommend my patch to `Registry` so that other pieces of Yard functionality I haven't come across yet but have a similar issue, have a chance of being fixed simultaneously.
